applications/tari_base_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -1825,6 +1826,69 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
 
         Ok(Response::new(response))
     }
+
+    async fn get_constitutions(
+        &self,
+        request: Request<tari_rpc::GetConstitutionsRequest>,
+    ) -> Result<Response<Self::GetConstitutionsStream>, Status> {
+        let report_error_flag = self.report_error_flag();
+        let request = request.into_inner();
+        let dan_node_public_key = PublicKey::from_bytes(&request.dan_node_public_key).map_err(|err| {
+            report_error(
+                report_error_flag,
+                Status::invalid_argument(format!("Dan node public key is not a valid public key:{}", err)),
+            )
+        })?;
+
+        let mut handler = self.node_service.clone();
+        let (mut sender, receiver) = mpsc::channel(50);
+        task::spawn(async move {
```

Comment:
nit: I wonder if it would be better to retrieve all the constitutions first, then spawn a task. That way you'd know what size to make the channel